src/test/scala/is/hail/io/LoadBgenSuite.scala

```diff
@@ -136,4 +151,71 @@ class LoadBgenSuite extends SparkSuite {
   @Test def testBgenImportRandom() {
     Spec.check()
   }
+
+  object TestProbIterator extends Properties("ImportBGEN") {
+    val bitPackedIteratorGen = for (
+      v <- Gen.buildableOf[Array, Double](Gen.choose(0d, 1d)).resize(10);
+      nBitsPerProb <- Gen.choose(1, 32)) yield (v, nBitsPerProb)
+
+    property("bgen probability iterator gives correct result") =
+      forAll(Gen.buildableOf[Array, Double](Gen.choose(0d, 1d)).resize(10)) { v =>
+        val probs = v.map(_ / v.sum)
+
+        (1 to 32).forall { nBitsPerProb =>
+
+          val probInts = probs.map(_ * (math.pow(2, nBitsPerProb) - 1))
+          val floorInts = probInts.map(i => (i, math.floor(i)))
+          val totalFractional = floorInts.map(f => f._1 - f._2).sum.toInt
+          val input = floorInts.sortBy(i => -(i._1 - i._2)).zipWithIndex.map { case (r, i) =>
+            if (i < totalFractional)
+              (i, math.ceil(r._1))
+            else
+              (i, math.floor(r._1))
+          }.map(_._2.toInt)
```

Comment:
I'll rewrite it as you suggest for this test. In the exportBGEN branch, the BGEN writer needs to do this to follow the rounding rules in the spec.